api/database/factories/ModelFactory.php

```diff
@@ -106,3 +107,20 @@
 
     return compact('token') + $body;
 });
+
+$factory->define(App\Models\ArticlePermalink::class, function ($faker) {
+    return [
+        'permalink' => str_random(255),
+    ];
+});
+
+$factory->define(App\Models\Article::class, function ($faker) {
+    $publishedDatetime = $faker->optional(0.9)->dateTimeThisDecade();
+    return [
+        'article_id' => $faker->uuid,
+        'title' => $faker->sentence,
+        'content' => $content = implode("\n\n", $faker->paragraphs(3)),
+        'permalink' => str_random(255),
```

Comment:
As above, but this should occasionally be null so `$faker->unique()->optional(0.9)->slug`